crates/mdb/src/error.rs

```diff
@@ -73,3 +58,22 @@ impl Error {
         unsafe { pchar_to_string(self.as_ref().message) }
     }
 }
+
+impl std::ops::Deref for BorrowedError<'_> {
+    type Target = Error;
+    // SAFETY: self.ptr is effectively a reference to Error provided by mdb-sys for the duration
```

Comment:
> self.ptr is effectively a reference to Error

This does not sound right to me. Given that

```rust
#[repr(transparent)]
struct `Error(*mut mdb_error_t)
```

I think `*const mdb_error_t` is more like `Error` than `&Error`. Am I missing something?

Can we back up the theory with a model implementation or mock implementation that we can run under miri?

I experimented a bit with modelling high risk patterns here: https://github.com/AxisCommunications/acap-rs/pull/140

Reply:
You are right, this was not correct. It should look like this:

```Rust
        unsafe { &*((&self.ptr as *const *const sys::error_t) as *const Error) }
```
That is to say, `self.ptr` is effectively a "const" `Error`, whatever that means. In this context though, since `Error` is transparently a `mdb_error_t`, we want to treat `&self.ptr` as a `&Error`. 

I validated this through miri, see the new commit.

Reply:
Yikes. But it looks correct as long as `self.ptr` does not move, which it doesn't.